src/main/java/org/elasticsearch/search/dfs/AggregatedDfs.java

```diff
@@ -82,20 +84,17 @@ public void readFrom(StreamInput in) throws IOException {
     @Override
     public void writeTo(final StreamOutput out) throws IOException {
         out.writeVInt(termStatistics.size());
-        final boolean[] states = termStatistics.allocated;
-        final Object[] keys = termStatistics.keys;
-        final Object[] values = termStatistics.values;
-        for (int i = 0; i < states.length; i++) {
-            if (states[i]) {
-                Term term = (Term) keys[i];
-                out.writeString(term.field());
-                out.writeBytesRef(term.bytes());
-                TermStatistics stats = (TermStatistics) values[i];
-                out.writeBytesRef(stats.term());
-                out.writeVLong(stats.docFreq());
-                out.writeVLong(DfsSearchResult.addOne(stats.totalTermFreq()));
-            }
+        
```

Comment:
I changed it to a cursor here; don't think this is crucial, but it can be reverted to a table scan, if needed (assuming no null keys are present, the scan is quite easy).